### ISSUE TITLE
Question

### ISSUE THREAD

**Issue by tanvirraihan142** [i6092467/vadesc#2]
Hi,

Thank you for developing this model and making it open source. My question is - 
Is there a way to test it on an entirely different dataset for which I do not have the event_time, event_observed variables and get a cluster_id output? In other words, I do not have the y_test and I want to get cluster_ids for a dataset with x_test only.

Thanks,
- R. B. Tanvir

**Comment by i6092467** [comment]
Hi @tanvirraihan142,

Thank you for your interest and question.

If you mean at the test time only, then yes, it is possible. I suggest checking out the `train.py` module. In brief, the way to do this is to set the `use_t` attribute to 0:
```
tf.keras.backend.set_value(model.use_t, np.array([0.0]))
rec, z_sample, p_z_c, p_c_z, risk_scores, lambdas = model.predict((x_test, y_test), batch_size=args.batch_size)
```
Note that `y_test` above can be a placeholder vector with any contents.

Of course, at training time, you will still need survival times and censoring indicators.

Let me know if you have further queries!

**Comment by tanvirraihan142** [comment]
Thanks for your answer. I did just check for using use_t set to 0 and 1. There seems to be a massive difference between predicting clusters for test set between use_t=0 and use_t=1. I wished that there would not be difference between those two cases but unfortunately thats not the case.

**Comment by i6092467** [comment]
The difference is expected and intended, see the bottom of p. 4 in the [original manuscript](https://openreview.net/pdf?id=RQ428ZptQfU) for technical details on how the cluster assignment is made when no $t$ is observed.

**Comment by tanvirraihan142** [comment]
I see. thanks for the clarification.
